examples/nnp_training.py

```diff
@@ -306,41 +313,51 @@ def validate():
     rmse = validate()
     print('RMSE:', rmse, 'at epoch', scheduler.last_epoch + 1)
 
-    learning_rate = optimizer.param_groups[0]['lr']
+    learning_rate = AdamW.param_groups[0]['lr']
 
     if learning_rate < early_stopping_learning_rate:
         break
 
-    tensorboard.add_scalar('validation_rmse', rmse, scheduler.last_epoch + 1)
-    tensorboard.add_scalar('best_validation_rmse', scheduler.best, scheduler.last_epoch + 1)
-    tensorboard.add_scalar('learning_rate', learning_rate, scheduler.last_epoch + 1)
-
     # checkpoint
     if scheduler.is_better(rmse, scheduler.best):
         torch.save(nn.state_dict(), best_model_checkpoint)
 
     scheduler.step(rmse)
 
-    for i, (batch_x, batch_y) in tqdm.tqdm(enumerate(training), total=len(training)):
+    tensorboard.add_scalar('validation_rmse', rmse, scheduler.last_epoch)
+    tensorboard.add_scalar('best_validation_rmse', scheduler.best, scheduler.last_epoch)
+    tensorboard.add_scalar('learning_rate', learning_rate, scheduler.last_epoch)
+
+    for i, (batch_x, batch_y) in tqdm.tqdm(
+        enumerate(training),
+        total=len(training),
+        desc="epoch {}".format(scheduler.last_epoch)
+    ):
+
         true_energies = batch_y['energies']
         predicted_energies = []
         num_atoms = []
+
         for chunk_species, chunk_coordinates in batch_x:
             num_atoms.append((chunk_species >= 0).to(true_energies.dtype).sum(dim=1))
             _, chunk_energies = model((chunk_species, chunk_coordinates))
             predicted_energies.append(chunk_energies)
-        num_atoms = torch.cat(num_atoms)
+
+        num_atoms = torch.cat(num_atoms).to(true_energies.dtype)
```

Comment:
Do we need the `.to(true_energies.dtype)`?

Reply:
No, fixed it in https://github.com/aiqm/torchani/pull/280/commits/cc8664a44b2e37cd8476a570fd605f65c615d274